src/s2/base/sysinfo.cc

```diff
@@ -15,15 +15,29 @@
 
 #include "s2/base/sysinfo.h"
 
-#include <sys/resource.h>
+#ifdef _MSC_VER /* if Visual C++ */
+#include <windows.h>
+#else
+#include <sys/resource.h>   // for rusage, RUSAGE_SELF
 #include <sys/time.h>
+#endif
 
 namespace base {
 
 absl::Duration CPUUsage() {
+#ifdef _MSC_VER /* if Visual C++ */
+  FILETIME creation, exit, kernel, user;
+  int res = GetProcessTimes(GetCurrentProcess(), &creation,
+                            &exit, &kernel, &user);
+  S2_CHECK_EQ(res, 0);
+  // high and low combine to 64-bit int, time uses 100-nanosecond intervals
+  return static_cast<double>((static_cast<uint64_t>(user.dwHighDateTime) << 32)
```

Comment:
Can you test this one final time? I think it needs to be `return absl::Duration(...);` now.